hivemind/optim/optimizer.py

```diff
@@ -524,8 +524,12 @@ def _begin_averaging_gradients(self, grad_scaler: Optional[GradScaler]) -> bool:
                 logger.exception(e)
 
         if not began_averaging_gradients and self.scheduled_grads is not None and not self.scheduled_grads.done():
-            logger.log(self.status_loglevel, f"Tagging along for a pre-scheduled gradient averaging round")
-            self._tag_along_with_zero_weight(self.scheduled_grads)
+            if self.tracker.global_progress.num_peers > 1:
+                logger.log(self.status_loglevel, f"Tagging along for a pre-scheduled gradient averaging round")
+                self._tag_along_with_zero_weight(self.scheduled_grads)
+            else:
+                logger.log(self.status_loglevel, f"Cancelling pre-scheduled averaging round: there are no other peers")
```

Comment:
```suggestion
                logger.log(self.status_loglevel, f"Skipping pre-scheduled averaging round: there are no other peers")
```

Because `Cancelling` sounds like we cancel an already started process abruptly.